maven-core/src/main/java/org/apache/maven/internal/transformation/TransformedArtifact.java

```diff
@@ -67,36 +71,36 @@ public void setFile(File file) {
 
     @Override
     public File getFile() {
-        Path result = onChangeTransformer.get();
-        if (result == null) {
-            return null;
+        try {
+            Path result = onChangeTransformer.transform();
+            if (result == null) {
+                return null;
+            }
+            return result.toFile();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
         }
-        return result.toFile();
     }
 
     private static final int BUFFER_SIZE = 8192;
 
-    private static String sha1(Path path) {
-        try {
-            MessageDigest md = MessageDigest.getInstance("SHA-1");
-            try (InputStream fis = Files.newInputStream(path)) {
-                byte[] buffer = new byte[BUFFER_SIZE];
-                int read;
-                while ((read = fis.read(buffer)) != -1) {
-                    md.update(buffer, 0, read);
-                }
-            }
-            StringBuilder result = new StringBuilder();
-            for (byte b : md.digest()) {
-                result.append(String.format("%02x", b));
+    private static String sha1(Path path) throws Exception {
```

Comment:
maybe IOException and perhaps another? Doesn't look like you have to throw  Exception here.